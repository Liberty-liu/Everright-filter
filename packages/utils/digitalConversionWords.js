import _ from 'lodash-es'
import Nzh from 'nzh'
import writtenNumber from 'written-number'
const digitalToChinese = (number) => {
  return Nzh.cn.encodeS(number)
}
const digitalToEnglish = (number) => {
  return _.startCase(writtenNumber(number))
}
export {
  digitalToChinese,
  digitalToEnglish
}
