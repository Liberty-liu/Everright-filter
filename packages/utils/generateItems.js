import _ from 'lodash-es'
const generateItems = (len = 1) => {
  const result = []
  while (len--) {
    result.push(_.uniqueId())
  }
  return result
}
export {
  generateItems
}
