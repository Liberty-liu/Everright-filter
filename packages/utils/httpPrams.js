import _ from 'lodash-es'
const apiParams = (id, method, defaultHttpParams, params = {}) => {
  const result = []
  const url = _.get(defaultHttpParams, `${id}.url`, '')
  if (url) {
    result[0] = url
  }
  const findParams = _.get(defaultHttpParams, `${id}.${method}`, '')
  if (findParams) {
    const data = {
      method
    }
    const key = method === 'get' ? 'query' : 'body'
    data[method === 'get' ? 'params' : 'data'] = findParams[key]
    result[1] = data
    _.merge(data, params)
  }
  return result
}
export {
  apiParams
}
