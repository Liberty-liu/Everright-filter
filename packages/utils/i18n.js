import _ from 'lodash-es'
const getLableByLang = (obj, lang) => {
  return lang === 'zh-cn' ? obj.label : (obj.en_label || obj.label)
}
const flatNodes = (nodes, fn) => {
  return nodes.reduce((res, node, currentIndex) => {
    res.push(node)
    fn && fn(nodes, node, currentIndex)
    const children = node.children || []
    res = res.concat(flatNodes(children, fn))
    return res
  }, [])
}
const convertDataByLang = (nodes, lang) => {
  const newNodes = _.cloneDeep(nodes)
  flatNodes(newNodes, (nodes, node) => {
    if (lang !== 'zh-cn') {
      node.label = node.en_label || node.label
    }
  })
  return newNodes
}
export {
  getLableByLang,
  convertDataByLang
}
