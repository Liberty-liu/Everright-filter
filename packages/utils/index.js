import _ from 'lodash-es'
const importModules = import.meta.glob('./*.js', { eager: true })
const modules = {}
_.forIn(importModules, (func, path) => {
  Object.keys(func).forEach((key) => {
    modules[key] = func[key]
  })
})
export default {
  ...modules
}
