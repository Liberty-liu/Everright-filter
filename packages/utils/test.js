const addTestId = (id, type = 'attr') => {
  let result = type === 'attr' ? {} : ''
  if (process.env.NODE_ENV === 'test') {
    if (type === 'attr') {
      result = { 'data-test-id': `er-filter-${id}` }
    }
    if (type === 'id') {
      result = `er-filter-${id}`
    }
  }
  // process.env.NODE_ENV === 'test' ? { 'data-test-id': `er-filter-${id}` } : {}
  return result
}
const getTestId = (id, type = 'attr') => type === 'attr' ? `[data-test-id="er-filter-${id}"]` : `.er-filter-${id}`
const addAttrs = (obj) => {
  return process.env.NODE_ENV === 'test' ? { 'data-test-attrs': JSON.stringify(obj) } : {}
}
const getAttrs = (el) => JSON.parse(el.getAttribute('data-test-attrs'))
export {
  addTestId,
  getTestId,
  addAttrs,
  getAttrs
}
