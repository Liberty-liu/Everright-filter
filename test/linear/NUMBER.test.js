import { describe, assert, expect, test, beforeEach, afterEach, beforeAll, vi } from 'vitest'
import { mount, flushPromises, enableAutoUnmount, config } from '@vue/test-utils'
import optionData from '@ER-server/routes/Filter/data/options.js'
import * as ElementPlusIconsVue from '@element-plus/icons-vue'
import NAME from '@ER/filter/name.js'
import utils from '@ER/utils'
import { nextTick } from 'vue'
import { getSelectOptions, _mount } from '../utils.js'
config.global.components.Delete = ElementPlusIconsVue.Delete
const getOptions = async () => {
  return new Promise((resolve, reject) => {
    resolve({
      data: optionData
    })
  })
}
describe('renderType: NUMBER', () => {
  let wrapper = {}
  const handleListener = vi.fn()
  beforeAll(async () => {
    wrapper = _mount(`
      <everright-filter
        :getOptions="getOptions"
        @listener="handleListener"
        ref="ERfilterRef"
      />
      `, () => ({
      handleListener,
      getOptions
    })
    )
  })
  afterEach(() => {
    wrapper.findComponent({ ref: 'ERfilterRef' }).vm.clearData()
  })
  test('operator: "style=noop" && value is not empty', async () => {
    await wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find('.Everright-filter-TriggerComponent').trigger('click')
    document.querySelector(utils.getTestId(`${NAME.TRIGGERCOMPONENT}-popperClass`, 'id')).querySelectorAll('.el-cascader-node')[0].click()
    await nextTick()
    document.querySelector(utils.getTestId(`${NAME.TRIGGERCOMPONENT}-popperClass`, 'id')).querySelectorAll('.el-cascader-menu')[1].querySelectorAll('.el-cascader-node')[6].click()
    await flushPromises()
    await wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId('operator', 'id')).trigger('click')
    expect(getSelectOptions(utils.getTestId('operator-popperClass', 'id')).map(e => utils.getAttrs(e).value)).toEqual([
      'equal',
      'not_equal',
      'greater_than',
      'greater_than_equal',
      'less_than',
      'less_than_equal',
      'between',
      'empty',
      'not_empty'
    ])
    let inputElm = wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId(`${NAME.NUMBERTYPE}-number0`, 'id'))
    const inputWrapper = inputElm.find('input')
    const nativeInput = inputWrapper.element
    expect(inputElm.exists()).toBe(true)
    expect(wrapper.findComponent({ ref: 'ERfilterRef' }).vm.getData()).toEqual({})
    await new Promise(resolve => setTimeout(resolve, 100))
    inputElm = wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId(`${NAME.NUMBERTYPE}-number0`, 'id'))
    expect(inputElm.classes()).toContain('ERFILTER-ERROR')
    nativeInput.value = '1'
    await inputWrapper.trigger('input')
    expect(wrapper.findComponent({ ref: 'ERfilterRef' }).vm.getData()).toMatchSnapshot()
  })
  test('operator: "style=range" && value is not empty', async () => {
    await wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find('.Everright-filter-TriggerComponent').trigger('click')
    document.querySelector(utils.getTestId(`${NAME.TRIGGERCOMPONENT}-popperClass`, 'id')).querySelectorAll('.el-cascader-node')[0].click()
    await nextTick()
    document.querySelector(utils.getTestId(`${NAME.TRIGGERCOMPONENT}-popperClass`, 'id')).querySelectorAll('.el-cascader-menu')[1].querySelectorAll('.el-cascader-node')[6].click()
    await flushPromises()
    await wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId('operator', 'id')).trigger('click')
    const selectOptions = getSelectOptions(utils.getTestId('operator-popperClass', 'id'))
    selectOptions[6].click()
    expect(wrapper.findComponent({ ref: 'ERfilterRef' }).vm.getData()).toEqual({})
    await new Promise(resolve => setTimeout(resolve, 100))
    const number0Elm = wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId(`${NAME.NUMBERTYPE}-number0`, 'id'))
    const number1Elm = wrapper.find(utils.getTestId(`${NAME.PICKERCOMPONENT}-0-0`)).find(utils.getTestId(`${NAME.NUMBERTYPE}-number1`, 'id'))
    expect(number0Elm.classes()).toContain('ERFILTER-ERROR')
    expect(number1Elm.classes()).toContain('ERFILTER-ERROR')
    number0Elm.find('input').element.value = '10'
    await number0Elm.find('input').trigger('input')
    number1Elm.find('input').element.value = '20'
    await number1Elm.find('input').trigger('input')
    expect(wrapper.findComponent({ ref: 'ERfilterRef' }).vm.getData()).toMatchSnapshot()
  })
})
