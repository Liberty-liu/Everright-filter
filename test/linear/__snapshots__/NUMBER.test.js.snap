// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renderType: NUMBER > operator: "style=noop" && value is not empty 1`] = `
{
  "filters": [
    {
      "conditions": [
        {
          "operator": "equal",
          "property": "rating",
          "value": 1,
        },
      ],
      "logicalOperator": "and",
    },
  ],
  "logicalOperator": "and",
}
`;
